### ISSUE TITLE
Add an example/guide showing how to solve a real-world problem 

### ISSUE THREAD

**Issue by crtomirmajer** [NiaOrg/NiaPy#215]
_This is a feature request._
Include an example or a guide of how to do ie. _feature selection_ using NiaPy. Not all of the potential users are familiar with the underlying algorithms and implementation details, but would still want to use this library to solve a problem (such as feature selection), using a more high-level API.
Here's a guide for feature selection from [PySwarms](https://pyswarms.readthedocs.io/en/latest/examples/feature_subset_selection.html) 

An example/guide could target something else also - I would just like to get an idea on how to solve a real-world problem using NiaPy, without going into details.

**Comment by sisco0** [comment]
Link is broken, it could be replaced by
[https://pyswarms.readthedocs.io/en/development/examples/feature_subset_selection.html](https://pyswarms.readthedocs.io/en/development/examples/feature_subset_selection.html)

**Comment by firefly-cpp** [comment]
Dear @crtomirmajer,

thank you very much for your interest in NiaPy. Please refer to the some interesting packages that are based on NiaPy:

https://github.com/karakatic/EvoPreprocess

https://github.com/lukapecnik/NiaAML

https://github.com/lukapecnik/NiaClass

https://github.com/timzatko/Sklearn-Nature-Inspired-Algorithms

**Comment by zStupan** [comment]
Hello everyone,
I've created a [gist](https://gist.github.com/zStupan/eee390bfc48e1ea2d8b9e2a0fbe05615), where I implemented SVM Hyper parameter optimization and feature selection using Particle Swarm optimization, based on [this paper](https://romisatriawahono.net/lecture/rm/paper/Lin%20-%20Parameter%20determination%20and%20feature%20selection%20for%20SVM%20by%20PSO%20-%202009.pdf).

I think it's correct, although it sometimes returns a worse model than the default. It says in the paper, they used 10 fold stratified cross validation. So I assumed they split the dataset into a training and test set and then did cross validation on the training set and returned the mean accuracy as the fitness function. That's how GridSearchCV works in scikit-learn. The article is not very clear about this. If anyone spots any mistakes in my code or something I missed from the article, please tell me.

I thought we could use this as the real world use case example for the docs, but I thing the best and easiest thing would be to take [this article](https://grega.xyz/2018/05/how-to-utilize-niapy-for-solving-knn-parameter-optimization-problem/) by @GregaVrbancic , throw it in an rst file and update the code to use the current API.
